frontend/test/resolution/testTuples.cpp

```diff
@@ -617,6 +617,71 @@ static void argHelper(std::string formal, std::string actual,
   }
 }
 
+static void test18() {
+  printf("test18\n");
+  Context ctx;
+  Context* context = &ctx;
+
+  auto program = R""""(
+                  var t = (1, "hello", 3.0);
+                  var x = t(0);
+                  var y = t(1);
+                  var z = t(2);
+                )"""";
+
+  auto m = parseModule(context, std::move(program));
+
+  auto x = findVariable(m, "x");
+  auto y = findVariable(m ,"y");
+  auto z = findVariable(m ,"z");
+
+  const ResolutionResultByPostorderID& rr = resolveModule(context, m->id());
+
+  assert(rr.byAst(x).type().type()->isIntType());
+  assert(rr.byAst(y).type().type()->isStringType());
+  assert(rr.byAst(z).type().type()->isRealType());
+}
+
+static void test19() {
+  printf("test19\n");
+  Context ctx;
+  Context* context = &ctx;
+
+  auto program = R""""(
+                  var t = (1, 2, 3);
+                  var x = t(0);
+                  var y = t(1);
+                  var z = t(2);
+
+                  var a = __primitive("get svec member", t, 0);
+                  var b = __primitive("get svec member value", t, 0);
```

Comment:
This seems sparsely tested for how much logic is in the primitive implementation. Both primitive return the same thing here -- should there be a case that distinguishes `svec` from `svec value`?

Reply:
There definitely should be and I've added one now.